third_party/terraform/tests/resource_compute_instance_test.go.erb

```diff
@@ -1876,6 +1876,53 @@ func TestAccComputeInstance_resourcePolicyCollocate(t *testing.T) {
 	})
 }
 
+func TestAccComputeInstance_subnetworkChange(t *testing.T) {
+	t.Parallel()
+
+	var instance compute.Instance
+	instanceName := fmt.Sprintf("tf-test-%s", randString(t, 10))
+	suffix := fmt.Sprintf("%s", randString(t, 10))
+	subnet := ""
+
+	checkAndUpdateSubnet := func(instance *compute.Instance) resource.TestCheckFunc {
+		oldSubnetwork := subnet
+		return func(s *terraform.State) error {
+			subnet = instance.NetworkInterfaces[0].Subnetwork
+			if oldSubnetwork != subnet {
+				return nil
+			}
+
+			return fmt.Errorf("subnetwork equivalent to previous... expected change; %s : %s", oldSubnetwork, subnet)
+		}
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeInstance_subnetworkChange(suffix, instanceName),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
Fun fact- most of these types of checks are completely unnecessary because of the way the test framework and the import steps work. The test framework automatically checks that, once applying the test config, if you were to run `terraform plan` again that it isn't empty. This makes sure that all the elements in state match the config- if they didn't, it would fail. The next part of it is the import step- this checks that if you had two environments, and you ran `terraform apply` in one and `terraform import` in the other, that the two states match. This ensures that we are correctly setting elements in state from the API, because if we didn't, the import would have the API value but the state would have the config value. So between those two checks, we can be confident that our configs are making the changes we expect they are. You'll see in most test files that we only do the import step as the check- this file is real old, so it probably was never cleaned up.

Reply:
cool ! Thanks for the information :)